src/App.jsx

```diff
@@ -1,17 +1,34 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+import messages from './data/messages.json';
 
 const App = () => {
+  const [chatMessages, setChatMessages] = useState(messages);
+
+  const toggleLike = (id) => {
+    const updatedMessages = chatMessages.map((message) => {
+      if (message.id === id) {
+        // message.liked = !message.liked;
+        return {...message, liked:!message.liked};
+      } else {
+        return message;
+      }
+    });
+    setChatMessages(updatedMessages);
+  };
```

Comment:
Nice work on this function! By passing this down to your individual chat messages you are now able to have a single source of truth!